### ISSUE TITLE
Overwritten transactions are not removed

### ISSUE THREAD

**Issue by monokh** [ethereum/eth-tester#57]
* Version: 0.1.0b15
* Python: 3.5
* OS: all

### What was wrong?

When submitting a transaction that overwrites another transaction, the original transaction is not removed and calls to `getTransaction()` still return the old transaction instead of null.

In geth, the transaction would be removed and returned as `None` from `getTransaction()`.

Example:

```
tester = EthereumTester(auto_mine_transactions=False)

web3 = Web3(EthereumTesterProvider(tester))

original_tx = web3.eth.sendTransaction({
    'from': '...',
    'to': '...',
    'value': 1,
    'gas': 23500,
    'gasPrice': 10,
    'nonce': 0
})

print(web3.eth.getTransaction(original_tx))

replacement_tx = web3.eth.sendTransaction({
    'from': '...',
    'to': '...',
    'value': 2,
    'gas': 23500,
    'gasPrice': 20,
    'nonce': 0
})

print(web3.eth.getTransaction(replacement_tx))
print(web3.eth.getTransaction(original_tx))  # Should be null but still returns transaction
```

### How can it be fixed?

Unsure whether this is a backend problem or not but i imagine there needs to be some tx pool managing functionality added.

**Comment by pipermerriam** [comment]
I'd consider this a bug that should be fixed such that it conforms to the geth behavior.

**Comment by carver** [comment]
Fixed by #60
